### PR TITLE
Fixes to relay stuff

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Probably best merged unsquashed.

### CODE REVIEW

File: cmd/strelaysrv/listener.go [hunk 1]
```diff
@@ -259,8 +258,8 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config) {
 			}
 			conn.Close()
 
-		case msg := <-outbox:
-			if msg == nil {
+		case msg, ok := <-outbox:
+			if !ok {
```
Comment (calmh) [inline cmd/strelaysrv/listener.go]: This will never happen if the outbox isn't closed?

Comment (calmh) [inline cmd/strelaysrv/listener.go]: Is the idea that the return always happens via the `<-errors` branch?

Comment (AudriusButkevicius) [inline cmd/strelaysrv/listener.go]: Yes

**Comment by AudriusButkevicius** [comment]
Addressed comment.

**Comment by calmh** [comment]
Merged

**Pull request closed without merging**